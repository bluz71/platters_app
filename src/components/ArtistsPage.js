import React, { Component } from 'react';
import axios from 'axios';
import { Row, Col, PageHeader } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import pluralize from 'pluralize';
import numeral from 'numeral';
import queryString from 'query-string';
import _ from 'lodash';
import { VelocityTransitionGroup } from 'velocity-react';
import { toast } from 'react-toastify';
import '../styles/ArtistsPage.css';
import { API_HOST } from '../config';
import pageProgress from '../helpers/pageProgress';
import Search from './Search';
import Paginator from './Paginator';
import ArtistsSidebar from './ArtistsSidebar';

const ARTISTS_ENDPOINT = `${API_HOST}/artists.json`;

class ArtistsPage extends Component {
  constructor(props) {
    super(props);

    // params is not React state since we do not want to re-render when it
    // changes; just use an instance variable instead.
    this.params       = {};
    this.loaded       = false;
    this.searching    = false;
    this.pageProgress = new pageProgress();

    this.state = {
      artists: [],
      mostRecentAlbums: [],
      mostRecentComments: [],
      pagination: {},
      error: null
    };

    // Bind 'this' for callback functions.
    this.handlePopState         = this.handlePopState.bind(this);
    this.handlePageChange       = this.handlePageChange.bind(this);
    this.handleAll              = this.handleAll.bind(this);
    this.handleLetter           = this.handleLetter.bind(this);
    this.handleSearchVisibility = this.handleSearchVisibility.bind(this);
    this.handleSearchChange     = this.handleSearchChange.bind(this);
    this.handleSearchSubmit     = this.handleSearchSubmit.bind(this);
  }

  componentDidMount() {
    window.onpopstate = this.handlePopState;
    this.applyState();
  }

  componentWillUnmount() {
    window.onpopstate = null;
  }

  shouldComponentUpdate(nextProps) {
    // Location change indicates either a direct 'this.props.history.push' or
    // back/forward in a browser. In both circumstances we do NOT want to
    // re-render since a GET for a new set of resources will immediately occur
    // after which will result in a re-render. Basically we only want to render
    // once upon a state change, not twice, hence we can safely ignore location
    // changes.
    return _.isEqual(this.props.location, nextProps.location);
  }

  handlePopState(event) {
    event.preventDefault();
    this.applyState();
  }

  handlePageChange(page) {
    const newParams = { ...this.params, page };
    this.applyParams(newParams);
  }

  handleAll() {
    const searching = this.searching;
    this.searching = false;
    const newParams = {};
    if (!this.applyParams(newParams) && searching) {
      // Hide the search form if it was previously visible.
      this.forceUpdate();
    }
  }

  handleLetter(letter) {
    const newParams = _.omit(this.params, ['page']);
    newParams.letter = letter;
    this.applyParams(newParams);
  }

  handleSearchVisibility() {
    this.searching = !this.searching;
    this.forceUpdate();
  }

  handleSearchChange(search) {
    const newParams = search ? { search } : {};
    this.applyParams(newParams);
  }

  handleSearchSubmit(event) {
    event.preventDefault();
    const search = this.search.value();
    const newParams = search ? { search } : {};
    this.applyParams(newParams);
  }

  // Apply state for back and forward transistion into/outof/within this
  // component.
  applyState() {
    if (this.props.location) {
      this.params = this.props.location.state || {};
    }
    else {
      this.params = {};
    }
    this.getArtists(false);
  }

  // Apply paramaters and update resources only if they are changed.
  applyParams(newParams) {
    if (_.isEqual(this.params, newParams)) {
      // Nothing to do, new params have not changed, just return.
      return false;
    }

    this.params = newParams;
    this.props.history.push('/artists', this.params);
    this.getArtists();
    return true;
  }

  artistsURL() {
    const params = queryString.stringify(this.params);

    return params.length > 0 ? `${ARTISTS_ENDPOINT}?${params}` : ARTISTS_ENDPOINT;
  }

  getArtists(scrollToTop = true) {
    axios.get(this.artistsURL())
      .then(response => {
        this.loaded = this.pageProgress.done();
        this.setState({
          artists: response.data.artists,
          mostRecentAlbums: response.data.most_recent.albums,
          mostRecentComments: response.data.most_recent.comments,
          pagination: response.data.pagination,
          error: null
        });
        if (scrollToTop) {
          window.scrollTo(0, 0);
        }
      })
      .catch(error => {
        this.pageProgress.done();
        this.setState({ error: error });
      });
  }

  letterActivity(letter) {
    if (this.params.letter === letter) {
      return 'active';
    }
  }

  artistsRetrieved() {
    if (!this.loaded || this.state.error) {
      return false;
    }
    else {
      return true;
    }
  }

  renderHeader() {
    const count = this.state.pagination.total_count;
    const artistsCount = numeral(count).format('0,0');

    return (
      <PageHeader>
        Artists {this.artistsRetrieved() && <small>({artistsCount} {pluralize('Artist', count)})</small>}
      </PageHeader>
    );
  }

  renderSearch() {
    return (
      <VelocityTransitionGroup
        enter={{
          animation: 'slideDown',
          duration: 250,
          complete: () => {
            this.search.focusSearchInput();
          }
        }}
        leave={{animation: 'slideUp', duration: 250}}>
        {this.searching &&
          <Search
            placeholder="Search Artists..."
            onSearchChange={this.handleSearchChange}
            onSearchSubmit={this.handleSearchSubmit}
            ref={(search) => this.search = search}
          />}
      </VelocityTransitionGroup>
    );
  }

  renderFilters() {
    if (!this.artistsRetrieved()) {
      return;
    }

    const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');

    return (
      <div>
        <div className="filters">
          <ul className="pagination pagination-sm">
            <li onClick={this.handleAll}><a>All</a></li>
            {!this.searching && letters.map((letter, index) => <li onClick={() => this.handleLetter(letter)} key={index} className={this.letterActivity(letter)}><a>{letter}</a></li>)}
            <li onClick={this.handleSearchVisibility}><FontAwesome name="search" /></li>
          </ul>
        </div>
        {this.renderSearch()}
      </div>
    );
  }

  renderArtists() {
    this.pageProgress.start();

    if (this.state.error) {
      toast.error('Connection failure, please retry again later', { className: 'ToastAlert' });
    }

    if (!this.state.error && this.loaded && this.state.artists.length === 0) {
      return <h4>No matching artists</h4>;
    }

    return (
      this.state.artists.map(artist =>
        <div key={artist.id} className="Artist">
          <Link to={`/artist/${artist.slug}`}>
            <h2>{artist.name}</h2>
            <p>{artist.description}</p>
            <span className="icon">
              <FontAwesome name="music" /> {artist.albums_count}
              <FontAwesome name="comment-o" className="spacer-left-xs" /> {artist.comments_count}
            </span>
          </Link>
        </div>
      )
    );
  }

  renderPaginator() {
    if (!this.state.error && this.state.pagination.total_pages > 1) {
      return (
        <Paginator pagination={this.state.pagination} onPageChange={this.handlePageChange} />
      );
    }
  }

  render() {
    return (
      <Row>
        <Col md={10}>
          <div className="Artists">
            {this.renderHeader()}
            {this.renderFilters()}
            {this.renderArtists()}
            {this.renderPaginator()}
          </div>
        </Col>
        <ArtistsSidebar
          mostRecentAlbums={this.state.mostRecentAlbums}
          mostRecentComments={this.state.mostRecentComments}
        />
      </Row>
    );
  }
}

export default ArtistsPage;
