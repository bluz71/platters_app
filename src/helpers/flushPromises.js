// Details:
//   https://blog.rescale.com/testing-promise-side-effects-with-asyncawait
//   https://github.com/facebook/jest/issues/2157#issuecomment-279171856

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

export default flushPromises;
