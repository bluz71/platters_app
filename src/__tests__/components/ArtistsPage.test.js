import React from 'react';
import { shallow } from 'enzyme';
import flushPromises from '../../helpers/flushPromises';
import ArtistsPage from '../../components/ArtistsPage';

describe('<ArtistsPage />', () => {
  it('renders without crashing', () => {
    shallow(<ArtistsPage />);
  });

  it('renders div.Artists', () => {
    const wrapper = shallow(<ArtistsPage />);
    expect(wrapper.find('div.Artists').length).toEqual(1);
  });

  it('renders first page', async () => {
    const wrapper = shallow(<ArtistsPage />);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders second page', async () => {
    const wrapper = shallow(<ArtistsPage history={[]} />);
    // Manually trigger the callback
    wrapper.instance().handlePageChange(2);
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders A artists', async () => {
    const wrapper = shallow(<ArtistsPage history={[]} />);
    // Manually trigger the callback
    wrapper.instance().handleLetter('A');
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ABC search match', async () => {
    const wrapper = shallow(<ArtistsPage history={[]} />);
    // Manually trigger the callback
    wrapper.instance().handleSearchChange('ABC');
    await flushPromises();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});

