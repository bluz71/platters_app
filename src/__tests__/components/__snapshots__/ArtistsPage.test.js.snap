// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistsPage /> renders A artists 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists
         
        <small>
          (
          1
           
          Artist
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              className="active"
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="1"
      >
        <Link
          replace={false}
          to="/artist/abc"
        >
          <h2>
            ABC
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            2
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            50
          </span>
        </Link>
      </div>
    </div>
  </Col>
  <ArtistsSidebar
    mostRecentAlbums={
      Array [
        Object {
          "album_slug": "abc",
          "artist": "ABC",
          "artist_slug": "abc",
          "cover_url": "http://localhost:3000/uploads/album/cover/898/small_ABC--ABC.jpg",
          "id": 1,
          "title": "ABC",
        },
        Object {
          "album_slug": "def",
          "artist": "DEF",
          "artist_slug": "def",
          "cover_url": "http://localhost:3000/uploads/album/cover/154/small_DEF--DEF.jpg",
          "id": 2,
          "title": "DEF",
        },
      ]
    }
    mostRecentComments={
      Array [
        Object {
          "created_at": "<time datetime=\\"2018-03-08T02:55:33Z\\" data-local=\\"time-ago\\">March  8, 2018  2:55am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 1,
          "name": "abc",
          "path": "artist/abc",
          "user_name": "fred",
          "user_slug": "fred",
        },
        Object {
          "created_at": "<time datetime=\\"2018-03-07T04:22:52Z\\" data-local=\\"time-ago\\">March  7, 2018  4:22am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 2,
          "name": "def",
          "path": "artist/def/album/def",
          "user_name": "jon",
          "user_slug": "jon",
        },
      ]
    }
  />
</Row>
`;

exports[`<ArtistsPage /> renders ABC search match 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists
         
        <small>
          (
          1
           
          Artist
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="1"
      >
        <Link
          replace={false}
          to="/artist/abc"
        >
          <h2>
            ABC
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            2
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            50
          </span>
        </Link>
      </div>
    </div>
  </Col>
  <ArtistsSidebar
    mostRecentAlbums={
      Array [
        Object {
          "album_slug": "abc",
          "artist": "ABC",
          "artist_slug": "abc",
          "cover_url": "http://localhost:3000/uploads/album/cover/898/small_ABC--ABC.jpg",
          "id": 1,
          "title": "ABC",
        },
        Object {
          "album_slug": "def",
          "artist": "DEF",
          "artist_slug": "def",
          "cover_url": "http://localhost:3000/uploads/album/cover/154/small_DEF--DEF.jpg",
          "id": 2,
          "title": "DEF",
        },
      ]
    }
    mostRecentComments={
      Array [
        Object {
          "created_at": "<time datetime=\\"2018-03-08T02:55:33Z\\" data-local=\\"time-ago\\">March  8, 2018  2:55am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 1,
          "name": "abc",
          "path": "artist/abc",
          "user_name": "fred",
          "user_slug": "fred",
        },
        Object {
          "created_at": "<time datetime=\\"2018-03-07T04:22:52Z\\" data-local=\\"time-ago\\">March  7, 2018  4:22am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 2,
          "name": "def",
          "path": "artist/def/album/def",
          "user_name": "jon",
          "user_slug": "jon",
        },
      ]
    }
  />
</Row>
`;

exports[`<ArtistsPage /> renders first page 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists
         
        <small>
          (
          3
           
          Artists
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="1"
      >
        <Link
          replace={false}
          to="/artist/abc"
        >
          <h2>
            ABC
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            2
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            50
          </span>
        </Link>
      </div>
      <div
        className="Artist"
        key="2"
      >
        <Link
          replace={false}
          to="/artist/def"
        >
          <h2>
            DEF
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            4
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            2
          </span>
        </Link>
      </div>
      <Paginator
        onPageChange={[Function]}
        pagination={
          Object {
            "current_page": 1,
            "next_page": 2,
            "prev_page": null,
            "total_count": 3,
            "total_pages": 2,
          }
        }
      />
    </div>
  </Col>
  <ArtistsSidebar
    mostRecentAlbums={
      Array [
        Object {
          "album_slug": "abc",
          "artist": "ABC",
          "artist_slug": "abc",
          "cover_url": "http://localhost:3000/uploads/album/cover/898/small_ABC--ABC.jpg",
          "id": 1,
          "title": "ABC",
        },
        Object {
          "album_slug": "def",
          "artist": "DEF",
          "artist_slug": "def",
          "cover_url": "http://localhost:3000/uploads/album/cover/154/small_DEF--DEF.jpg",
          "id": 2,
          "title": "DEF",
        },
      ]
    }
    mostRecentComments={
      Array [
        Object {
          "created_at": "<time datetime=\\"2018-03-08T02:55:33Z\\" data-local=\\"time-ago\\">March  8, 2018  2:55am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 1,
          "name": "abc",
          "path": "artist/abc",
          "user_name": "fred",
          "user_slug": "fred",
        },
        Object {
          "created_at": "<time datetime=\\"2018-03-07T04:22:52Z\\" data-local=\\"time-ago\\">March  7, 2018  4:22am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 2,
          "name": "def",
          "path": "artist/def/album/def",
          "user_name": "jon",
          "user_slug": "jon",
        },
      ]
    }
  />
</Row>
`;

exports[`<ArtistsPage /> renders second page 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={10}
  >
    <div
      className="Artists"
    >
      <PageHeader
        bsClass="page-header"
      >
        Artists
         
        <small>
          (
          3
           
          Artists
          )
        </small>
      </PageHeader>
      <div>
        <div
          className="filters"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              onClick={[Function]}
            >
              <a>
                All
              </a>
            </li>
            <li
              key="0"
              onClick={[Function]}
            >
              <a>
                A
              </a>
            </li>
            <li
              key="1"
              onClick={[Function]}
            >
              <a>
                B
              </a>
            </li>
            <li
              key="2"
              onClick={[Function]}
            >
              <a>
                C
              </a>
            </li>
            <li
              key="3"
              onClick={[Function]}
            >
              <a>
                D
              </a>
            </li>
            <li
              key="4"
              onClick={[Function]}
            >
              <a>
                E
              </a>
            </li>
            <li
              key="5"
              onClick={[Function]}
            >
              <a>
                F
              </a>
            </li>
            <li
              key="6"
              onClick={[Function]}
            >
              <a>
                G
              </a>
            </li>
            <li
              key="7"
              onClick={[Function]}
            >
              <a>
                H
              </a>
            </li>
            <li
              key="8"
              onClick={[Function]}
            >
              <a>
                I
              </a>
            </li>
            <li
              key="9"
              onClick={[Function]}
            >
              <a>
                J
              </a>
            </li>
            <li
              key="10"
              onClick={[Function]}
            >
              <a>
                K
              </a>
            </li>
            <li
              key="11"
              onClick={[Function]}
            >
              <a>
                L
              </a>
            </li>
            <li
              key="12"
              onClick={[Function]}
            >
              <a>
                M
              </a>
            </li>
            <li
              key="13"
              onClick={[Function]}
            >
              <a>
                N
              </a>
            </li>
            <li
              key="14"
              onClick={[Function]}
            >
              <a>
                O
              </a>
            </li>
            <li
              key="15"
              onClick={[Function]}
            >
              <a>
                P
              </a>
            </li>
            <li
              key="16"
              onClick={[Function]}
            >
              <a>
                Q
              </a>
            </li>
            <li
              key="17"
              onClick={[Function]}
            >
              <a>
                R
              </a>
            </li>
            <li
              key="18"
              onClick={[Function]}
            >
              <a>
                S
              </a>
            </li>
            <li
              key="19"
              onClick={[Function]}
            >
              <a>
                T
              </a>
            </li>
            <li
              key="20"
              onClick={[Function]}
            >
              <a>
                U
              </a>
            </li>
            <li
              key="21"
              onClick={[Function]}
            >
              <a>
                V
              </a>
            </li>
            <li
              key="22"
              onClick={[Function]}
            >
              <a>
                W
              </a>
            </li>
            <li
              key="23"
              onClick={[Function]}
            >
              <a>
                X
              </a>
            </li>
            <li
              key="24"
              onClick={[Function]}
            >
              <a>
                Y
              </a>
            </li>
            <li
              key="25"
              onClick={[Function]}
            >
              <a>
                Z
              </a>
            </li>
            <li
              onClick={[Function]}
            >
              <FontAwesome
                name="search"
              />
            </li>
          </ul>
        </div>
        <VelocityTransitionGroup
          enter={
            Object {
              "animation": "slideDown",
              "complete": [Function],
              "duration": 250,
            }
          }
          enterHideStyle={
            Object {
              "display": "none",
            }
          }
          enterShowStyle={
            Object {
              "display": "",
            }
          }
          leave={
            Object {
              "animation": "slideUp",
              "duration": 250,
            }
          }
          runOnMount={false}
        />
      </div>
      <div
        className="Artist"
        key="3"
      >
        <Link
          replace={false}
          to="/artist/ghi"
        >
          <h2>
            GHI
          </h2>
          <p />
          <span
            className="icon"
          >
            <FontAwesome
              name="music"
            />
             
            7
            <FontAwesome
              className="spacer-left-xs"
              name="comment-o"
            />
             
            23
          </span>
        </Link>
      </div>
      <Paginator
        onPageChange={[Function]}
        pagination={
          Object {
            "current_page": 2,
            "next_page": null,
            "prev_page": 1,
            "total_count": 3,
            "total_pages": 2,
          }
        }
      />
    </div>
  </Col>
  <ArtistsSidebar
    mostRecentAlbums={
      Array [
        Object {
          "album_slug": "abc",
          "artist": "ABC",
          "artist_slug": "abc",
          "cover_url": "http://localhost:3000/uploads/album/cover/898/small_ABC--ABC.jpg",
          "id": 1,
          "title": "ABC",
        },
        Object {
          "album_slug": "def",
          "artist": "DEF",
          "artist_slug": "def",
          "cover_url": "http://localhost:3000/uploads/album/cover/154/small_DEF--DEF.jpg",
          "id": 2,
          "title": "DEF",
        },
      ]
    }
    mostRecentComments={
      Array [
        Object {
          "created_at": "<time datetime=\\"2018-03-08T02:55:33Z\\" data-local=\\"time-ago\\">March  8, 2018  2:55am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 1,
          "name": "abc",
          "path": "artist/abc",
          "user_name": "fred",
          "user_slug": "fred",
        },
        Object {
          "created_at": "<time datetime=\\"2018-03-07T04:22:52Z\\" data-local=\\"time-ago\\">March  7, 2018  4:22am</time>",
          "gravatar_url": "https://gravatar.com/avatar/88884e5cb90c02031f031bc4760fcf69?s=80&r=pg&d=identicon",
          "id": 2,
          "name": "def",
          "path": "artist/def/album/def",
          "user_name": "jon",
          "user_slug": "jon",
        },
      ]
    }
  />
</Row>
`;
